### ISSUE TITLE
The 'bbox_iou' function on line 177 in 'yolov4.py' seems wrong

### ISSUE THREAD

**Issue by TCBocean** [hunglc007/tensorflow-yolov4-tflite#24]
https://github.com/hunglc007/tensorflow-yolov4-tflite/blob/9a4f98ea0276da08a69e548010562e68b940b5b9/core/yolov4.py#L187
The box1 and box1 of this line of code should not be able to get the coordinates of the upper left corner of the overlapping area.

**Comment by zxj11838** [comment]
I also found the same problem, this may be the reason why the network training loss is nan

**Comment by robisen1** [comment]
> I also found the same problem, this may be the reason why the network training loss is nan

do you have a solution to the issue?

**Comment by zxj11838** [comment]
I have changed，
left_up = tf.maximum(boxes1_coor[..., :2], boxes1_coor[..., :2])
to：
left_up = tf.maximum(boxes1_coor[..., :2], boxes2_coor[..., :2])

But the loss of training is still nan.

**Comment by rubencardenes** [comment]
Yes, that function is definitely wrong. I changed that line to compute left_up coordinates, as well as the right_down coordinates. I also found in some implementations the iou is computed using the maximum between the value and an epsilon as this:

```
def bbox_iou(boxes1, boxes2):

    boxes1_area = boxes1[..., 2] * boxes1[..., 3]
    boxes2_area = boxes2[..., 2] * boxes2[..., 3]

    boxes1_coor = tf.concat([boxes1[..., :2] - boxes1[..., 2:] * 0.5,
                             boxes1[..., :2] + boxes1[..., 2:] * 0.5], axis=-1)
    boxes2_coor = tf.concat([boxes2[..., :2] - boxes2[..., 2:] * 0.5,
                             boxes2[..., :2] + boxes2[..., 2:] * 0.5], axis=-1)

    left_up    = tf.maximum(boxes1_coor[..., :2], boxes2_coor[..., :2])
    right_down = tf.minimum(boxes1_coor[..., 2:], boxes2_coor[..., 2:])

    inter_section = tf.maximum(right_down - left_up, 0.0)
    inter_area = inter_section[..., 0] * inter_section[..., 1]
    union_area = boxes1_area + boxes2_area - inter_area
    iou       = tf.maximum(1.0 * inter_area / union_area, np.finfo(np.float32).eps)

    return iou
```
